pokedex-with-bit/executors.py

```diff
@@ -0,0 +1,317 @@
+        d_emb = _ext_B(_norm(b))
+        dists = _cosine(q_emb, d_emb)
+        top_k = int(parameters.get('top_k', 5))
+        idx, dist = self._get_sorted_top_k(dists, top_k)
+        for _q, _ids, _dists in zip(docs, idx, dist):
+            for _id, _dist in zip(_ids, _dists):
+                doc = Document(self._docs[int(_id)], copy=True)
+                doc.score.value = 1 - _dist
+                doc.parent_id = int(_id)
+                _q.matches.append(doc)
+        return docs
+
+    @staticmethod
+    def _get_sorted_top_k(
+        dist: 'np.array', top_k: int
+    ) -> Tuple['np.ndarray', 'np.ndarray']:
+        if top_k >= dist.shape[1]:
+            idx = dist.argsort(axis=1)[:, :top_k]
+            dist = np.take_along_axis(dist, idx, axis=1)
+        else:
+            idx_ps = dist.argpartition(kth=top_k, axis=1)[:, :top_k]
+            dist = np.take_along_axis(dist, idx_ps, axis=1)
+            idx_fs = dist.argsort(axis=1)
+            idx = np.take_along_axis(idx_ps, idx_fs, axis=1)
+            dist = np.take_along_axis(dist, idx_fs, axis=1)
+
+        return idx, dist
+
+
+class KeyValueIndexer(Executor):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if os.path.exists(self.save_path):
+            self._docs = DocumentArray.load(self.save_path)
+        else:
+            self._docs = DocumentArray()
+
+    @property
+    def save_path(self):
+        if not os.path.exists(self.workspace):
+            os.makedirs(self.workspace)
+        return os.path.join(self.workspace, 'kv.json')
+
+    def close(self):
+        self._docs.save(self.save_path)
+
+    @requests(on='/index')
+    def index(self, docs: DocumentArray, **kwargs):
+        self._docs.extend(docs)
+
+    @requests(on='/search')
+    def query(self, docs: DocumentArray, **kwargs):
+        for doc in docs:
+            for match in doc.matches:
+                extracted_doc = self._docs[int(match.parent_id)]
+                # The id fields should be the same
+                assert match.id == extracted_doc.id
+                match.MergeFrom(extracted_doc)
+        return docs
+
+
+class MatchImageReader(Executor):
+    def __init__(
+        self,
+        target_size: Union[Iterable[int], int] = 224,
+        img_mean: Tuple[float] = (0, 0, 0),
+        img_std: Tuple[float] = (1, 1, 1),
+        resize_dim: int = 256,
+        channel_axis: int = -1,
+        target_channel_axis: int = -1,
+        *args,
+        **kwargs,
+    ):
+        """Set Constructor."""
+        super().__init__(*args, **kwargs)
+        self.target_size = target_size
+        self.resize_dim = resize_dim
+        self.img_mean = np.array(img_mean).reshape((1, 1, 3))
+        self.img_std = np.array(img_std).reshape((1, 1, 3))
+        self.channel_axis = channel_axis
+        self.target_channel_axis = target_channel_axis
+
+    @requests(on='/search')
+    def query(self, docs: DocumentArray, **kwargs):
```

Comment:
please add type hint for return value

Reply:
https://github.com/jina-ai/examples/pull/626/commits/131bee3152cf88e330cc3b9856122a596ca984dd